app.py

```diff
@@ -19,23 +20,58 @@ def exportBag():
     "sample.tsv" in its "data" sub-directory. It creates
     a new workspace in a FireCloud namespace, and uploads
     the data into it.
+    Aborts execution and returns HTTP body back to client if any 
+    request operation fails to succeed.
     """
     req_body = app.current_request.raw_body
     req_query_params = app.current_request.query_params
     req_headers = app.current_request.headers
     os.chdir('/tmp')
     with zipfile.ZipFile(io.BytesIO(req_body), 'r') as archive:
         archive.extractall()
-    payload = []
-    payload.append('/tmp/manifest/data/participant.tsv')
-    payload.append('/tmp/manifest/data/sample.tsv')
+    data = {
+        'participant': '/tmp/manifest/data/participant.tsv',
+        'sample': '/tmp/manifest/data/sample.tsv'
+    }
+    url = 'https://api.firecloud.org/api/workspaces'
     workspace = req_query_params['workspace']
     namespace = req_query_params['namespace']
-    url = 'https://api.firecloud.org/api/workspaces'
     auth = req_headers['Authorization']
-    manifest_io = ManifestIO(payload, url, workspace, namespace, auth)
-    statusCode = manifest_io.import_tsv_to_fc()
-    return Response(body=statusCode,
-                    status_code=200,
+    # Dict to hold status codes (or booleans) for successful operation.
+    statusCode = {
+        'status codes': {
+            'workspace exists': '',
+            'stood up workspace': '',
+            'participant upload': '',
+            'sample upload': ''
+        }
+    }
+    manifest_io = ManifestIO(data, url, workspace, namespace, auth, statusCode)
+    # Check whether the workspace exists in this namespace, otherwise create it.
+    if not manifest_io.workspace_exists():
+        response = manifest_io.standup_workspace()
+        if response.status_code == 201:
+            manifest_io.statusCode['status codes']['stood up workspace'] = \
+                response.status_code
+        else:
+            return Response(body=response.text,
+                            headers={'Content-Type': 'text/plain',
+                                     'Accept': 'application/json'})
```

Comment:
Accept header is only for requests, not responses, so you should remove that.

Ideally you would return the FC response, so if they returned a JSON, you should return that.

Reply:
I'll take out the header. WRT your next point, I return the structure FC returns, i.e., my return content types matches FC's type in the HTTP body.

Reply:
The content type of in `headers` of the return-to-client `Response` is now the content type in the headers from the FC response ("proxy"). A short utility returns the value for the `Resonse.body`: if FC returned JSON it will be JSON, otherwise it will be `text/plain` as in responses other than 200 or 404 the response has no JSON and throws an error if you try to call it.